src/services/suggestionDiagnostics.ts

```diff
@@ -17,12 +17,27 @@ namespace ts {
             switch (node.kind) {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.FunctionExpression:
+
                     if (isJsFile) {
                         const symbol = node.symbol;
                         if (symbol.members && (symbol.members.size > 0)) {
                             diags.push(createDiagnosticForNode(isVariableDeclaration(node.parent) ? node.parent.name : node, Diagnostics.This_constructor_function_may_be_converted_to_a_class_declaration));
                         }
                     }
+                    
+                    if(node.modifiers && containsAsync(node.modifiers)){
+                        break;
+                    }
+
+                    const returnType = checker.getReturnTypeOfSignature(checker.getSignatureFromDeclaration(<FunctionDeclaration | FunctionExpression> node))
+                    if(!returnType || !returnType.symbol){
+                        break;
+                    }
+
+                    if(isPromiseType(returnType)){

```

Comment:
i am not sure this is sufficient check to be able convert it. e.g.:
```ts
function f() {
    return g();
}

async function g() { return 0; }
```

`f` returns a `Promise`, but there is no reason why this should be converted to an async function.

I think a better way to look at this is not the `async` part but rather the `await` part. if the function includes a possible use of `await`. in other words, there a call to `.then` on a promise in the function.

Reply:
since we are talking about it should be the use of any of  `.then`, `.catch` and `.finally`